metrics/web/test/dashboard/presentation/state/project_metrics_notifier_test.dart

```diff
@@ -337,15 +341,15 @@ void main() {
       final expectedProjectMetrics =
           projectMetricsNotifier.projectsMetricsTileViewModels;
 
-      final listener = expectAsync0(() {
-        final filteredProjectMetrics =
-            projectMetricsNotifier.projectsMetricsTileViewModels;
+      final listener = expectAsyncUntil0(
+        () {},
+        () {
+          final filteredProjectMetrics =
+              projectMetricsNotifier.projectsMetricsTileViewModels;
+          return listEquals(filteredProjectMetrics, expectedProjectMetrics);
+        },
+      );
 
-        expect(
-          filteredProjectMetrics,
-          equals(expectedProjectMetrics),
-        );
-      });
       projectMetricsNotifier.addListener(listener);
       projectMetricsNotifier.filterByProjectName(null);
     });
```

Comment:
I think we should add a comma here and format the code, to make the alignment look better.

```suggestion
    },);
```